src/components/PendingTxsSnackbar/store.test.js

```diff
@@ -0,0 +1,116 @@
+import { TransactionType, TransactionStatus } from 'constants';
+import { observable } from 'mobx';
+import PendingTxsSnackbarStore from './store';
+import { mockResetTransactionList, mockAddTransaction } from '../../network/graphql/queries/';
+
+jest.mock('../../network/graphql/queries'); // block and manually mock our backend
+
+describe('PendingTxsSnackbarStore', () => {
+  let store;
+  const app = {
+    global: observable({ syncBlockNum: 0, syncPercent: 10 }),
+  }; // mock the appstore
+  const sleep = ms => new Promise(res => setTimeout(res, ms));
+
+  describe('Init', () => {
+    beforeEach(() => {
+      store = new PendingTxsSnackbarStore(app); // create a new instance before each test case
+      mockResetTransactionList();
+    });
+    it('Init with null tx list', async () => {
+      await store.init();
+      expect(store.count).toBe(0);
+      expect(store.isVisible).toBe(false);
+      expect.assertions(2);
+    });
+    it('Init with tx list', async () => {
+      await store.queryPendingTransactions();
+      expect(store.count).toBe(0);
+      expect(store.isVisible).toBe(false);
+      mockAddTransaction({ txid: 1, status: TransactionStatus.PENDING, type: TransactionType.APPROVE_CREATE_EVENT });
+      await store.queryPendingTransactions();
+      expect(store.count).toBe(1);
+      expect(store.pendingCreateEvents.length).toBe(1);
+      expect(store.isVisible).toBe(true);
+      mockAddTransaction({ txid: 2, status: TransactionStatus.PENDING, type: TransactionType.CREATE_EVENT });
+      await store.queryPendingTransactions();
+      expect(store.count).toBe(2);
+      expect(store.pendingCreateEvents.length).toBe(2);
+      expect(store.isVisible).toBe(true);
+      mockAddTransaction({ txid: 3, status: TransactionStatus.PENDING, type: TransactionType.BET });
+      await store.queryPendingTransactions();
+      expect(store.count).toBe(3);
+      expect(store.pendingCreateEvents.length).toBe(2);
+      expect(store.pendingBets.length).toBe(1);
+      expect(store.isVisible).toBe(true);
+      mockResetTransactionList();
```

Comment:
The purpose of mockResetTransactionList() is to remove some transactions from the query result, not to mimic the DB resetting.
I'm not sure whether my knowledge is correct, but I think the total amount of the query results can decreases if some transactions changes their status. e.g. if a transaction change their status from pending to success, it will not appeared at a query for all pending transaction anymore.